device/globals/global_vm.gd

```diff
@@ -131,11 +132,76 @@ func settings_loaded(p_settings):
 func music_volume_changed():
 	emit_signal("music_volume_changed")
 
+func hover_debug(s):
+	var ids = []
+	for h in hover_stack:
+		ids.push_back([h.z_index, h.global_id])
+	printt("HOVER DEBUG: " + s, ids)
+
+func hover_push(obj):
+	if not obj is esc_type.ITEM and not obj is esc_type.TRIGGER:
+		report_errors("global_vm", ["Trying to hover " + obj.global_id + " which is not ITEM or TRIGGER"])
+
+	var stacked
+	var for_else = true
+	var need_new_hover = false
+	if not hover_stack:
+		hover_stack.push_back(obj)
+		need_new_hover = true
+	else:
+		if obj in hover_stack:
+			report_errors("global_vm", ["Hovering push obj " + obj.global_id + " in hover_stack!"])
+
+		for i in hover_stack.size():
+			stacked = hover_stack[i]
+			if obj.z_index >= stacked.z_index:
+				hover_stack.insert(i, obj)
+				for_else = false
+				need_new_hover = i == 0
+				break
+		if for_else:
+			hover_stack.push_back(obj)
+
+	hover_debug("PUSH")
+	if need_new_hover:
+		# hover_debug("PUSHED, hovering " + obj.global_id)
+		hover_begin(obj)
+	# else:
+	# 	hover_debug("KEEP HOVERING " + hover_object.global_id)
+
+func hover_pop(obj):
+	if not obj in hover_stack:
+		report_errors("global_vm", ["Hovering pop obj " + obj.global_id + " not in hover_stack!"])
+
+	if not hover_stack:
+		report_errors("global_vm", ["Hovering trying to pop from empty stack"])
+
+	var next
+	if hover_stack[0] == obj:
+		hover_stack.pop_front()
+		if hover_stack:
+			next = hover_stack[0]
+	else:
+		for i in range(1, hover_stack.size()):
+			if hover_stack[i] == obj:
+				next = hover_stack[i - 1]
+				hover_stack.remove(i)
+				break
+
+	hover_debug("POP")
```

Comment:
Yeah, is it ok if I commit with it commented out? To show where and how to debug?

As an aside, I'm not personally a fan of that convention, but because there are no logging facilities and logging is a bit chatty and so on, commenting out and showing where to comment back in _might_ be the best approach.

Is there a simple way for Escoria to create new panes or tabs in the Output/Debugger pane for different types of output?